docs/src/references/agent/operation-workflow.md

```diff
@@ -286,4 +288,165 @@ next = ["restarting", "successful_restart", "failed_restart"]
+on_exit._ = "failed"                                      # wildcard for any other non successful exit
+on_kill = { status = "failed", reason = "killed"}         # next status when killed
+```
+
+- `on_success` is syntactic sugar for `on_exit.0`
+- `on_error` is syntactic sugar for `on_exit._`
+- This is an error to provide more than one handler for an exit code (overlapping ranges).
+- If no reason is provided for an exit code, the default reason is `"${program} exited with ${code}"`.
+- If no reason is provided for the on kill handler, the reason is `"${program} killed by ${code}"`.
+- The default for the `on-error` and `on-kill` status is `"failed"`.
+- There is no default for the `on-success` status.
+  In such a case the output of the script is used to determine the next status.
+
+If the standard output of the script contains a JSON object surrounded by `:::begin-tedge:::` and `:::end-tedge:::` markers,
+then this object is injected in the command state message payload.
+There are two exceptions, though.
+The `status` and `reason` fields are determined after the exit code and not the standard output.
+
+### Next step determined by script output
+
+The output of the script processing a command state
+can be used to determine the next state of the workflow.
+
+For that to work:
+- no `on_success` nor `on_exit.0` status must be given as this would make the next status computed after the exit code
+- the standard output of the script must emit a JSON object surrounded by `:::begin-tedge:::` and `:::end-tedge:::` markers.
+- this JSON object must provide a `status` field and possibly a `reason` field.
+
+```toml
+script = "/some/script.sh with some args"                                # no given `on_exit.0` status
+on_error = { status = "fatal_state", reason = "fail to run the script"}  # possibly some `on_error` and `on_kill` handlers
+next = ["state-1", "state-2", "state-3"]                                 # the list of status accepted as next status
+```
+
+- If the script is successful and its output returns some `status` and `reason` fields, these are used for the next state.
+- If the script is successful but its output contains no `status` field, then `on_error` is used.
+- If the script cannot be executed, is killed or returns a non-zero code,
+  then the `status` and `reason` fields are determined by the `on_error` and `on_kill` handlers
+  (or if none where provided using the default `on_error` and `on_kill` handlers).
+- The fields of the JSON object extracted from the script output are injected into the command state message payload
+  - This is done in the successful as well as the failed cases,
+    for all the fields, except for the `status` and `reason` fields.
+  - Notably, when the script returns with a non-successful status code,
+    the `on_error` definition trumps over any `status` and `reason` fields provided over the script stdout.
```

Comment:
Yes, definitely.

- Currently, even stronger. If the script is non successful its output is ignored.
- But maybe this is not what you have in mind. Do you think consuming the output can also be useful in case of exit code with a specific handler? Somehow these are expected errors.

Reply:
After discussion, it appears that things are a bit more subtle:

- the `status` given in the workflow definition trumps any `status` value published on stdout (the idea is that sequence of states is statically defined)
- the `reason` published on stdout trumps the `reason` provided in the workflow definition (the rational being that an error message is dynamically provided by the running process).
- these rules applies only for the exit codes which are somehow expected .i.e. with a defined `on_exit` handler.